internal/mode/static/state/graph/httproute.go

```diff
@@ -382,40 +389,99 @@ func validateFilterHeaderModifierFields(
 	// Ensure that the header names are case-insensitive unique
 	allErrs = append(allErrs, validateRequestHeadersCaseInsensitiveUnique(
 		headerModifier.Add,
-		headerModifierPath.Child("add"))...,
+		headerModifierPath.Child(add))...,
 	)
 	allErrs = append(allErrs, validateRequestHeadersCaseInsensitiveUnique(
 		headerModifier.Set,
-		headerModifierPath.Child("set"))...,
+		headerModifierPath.Child(set))...,
 	)
 	allErrs = append(allErrs, validateRequestHeaderStringCaseInsensitiveUnique(
 		headerModifier.Remove,
-		headerModifierPath.Child("remove"))...,
+		headerModifierPath.Child(remove))...,
 	)
 
 	for _, h := range headerModifier.Add {
-		if err := validator.ValidateRequestHeaderName(string(h.Name)); err != nil {
-			valErr := field.Invalid(headerModifierPath.Child("add"), h, err.Error())
+		if err := validator.HTTPHeaderValidator(string(h.Name)); err != nil {
+			valErr := field.Invalid(headerModifierPath.Child(add), h, err.Error())
 			allErrs = append(allErrs, valErr)
 		}
-		if err := validator.ValidateRequestHeaderValue(h.Value); err != nil {
-			valErr := field.Invalid(headerModifierPath.Child("add"), h, err.Error())
+		if err := validator.ValidateFilterHeaderValue(h.Value); err != nil {
+			valErr := field.Invalid(headerModifierPath.Child(add), h, err.Error())
 			allErrs = append(allErrs, valErr)
 		}
 	}
 	for _, h := range headerModifier.Set {
-		if err := validator.ValidateRequestHeaderName(string(h.Name)); err != nil {
-			valErr := field.Invalid(headerModifierPath.Child("set"), h, err.Error())
+		if err := validator.HTTPHeaderValidator(string(h.Name)); err != nil {
+			valErr := field.Invalid(headerModifierPath.Child(set), h, err.Error())
 			allErrs = append(allErrs, valErr)
 		}
-		if err := validator.ValidateRequestHeaderValue(h.Value); err != nil {
-			valErr := field.Invalid(headerModifierPath.Child("set"), h, err.Error())
+		if err := validator.ValidateFilterHeaderValue(h.Value); err != nil {
+			valErr := field.Invalid(headerModifierPath.Child(set), h, err.Error())
 			allErrs = append(allErrs, valErr)
 		}
 	}
 	for _, h := range headerModifier.Remove {
-		if err := validator.ValidateRequestHeaderName(h); err != nil {
-			valErr := field.Invalid(headerModifierPath.Child("remove"), h, err.Error())
+		if err := validator.HTTPHeaderValidator(h); err != nil {
+			valErr := field.Invalid(headerModifierPath.Child(remove), h, err.Error())
+			allErrs = append(allErrs, valErr)
+		}
+	}
+
+	return allErrs
+}
+
+func validateFilterResponseHeaderModifier(
+	validator validation.HTTPFieldsValidator,
+	responseHeaderModifier *v1.HTTPHeaderFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	if errList := validateFilterHeaderModifier(validator, responseHeaderModifier, filterPath); errList != nil {
+		return errList
+	}
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, validateResponseHeaders(
```

Comment:
nit: the line breaks with the append statements are not quite right

I would write it as:
```
allErrs = append(
    allErrs, 
    validateResponseHeaders(responseHeaderModifier.Add,filterPath.Child(add))...,
)
```